src/demos/findex/findex.ts

```diff
@@ -0,0 +1,172 @@
+import { AbeMasterKey } from "../../crypto/abe/keygen/keygen";
+import { Policy, PolicyAxis } from "../../crypto/abe/keygen/policy";
+import { generateMasterKeys, coverCryptEncrypt, coverCryptDecrypt } from "../abe/cover_crypt/cover_crypt";
+import { Findex } from "../../interface/findex/findex";
+import { DB } from "./demo_db";
+import { masterKeysFindex } from "./demo_keys";
+import { logger } from "../../utils/logger";
+import { hexDecode, hexEncode, sanitizeString } from "../../utils/utils";
+
+export class FindexDemo {
+  private _db: DB;
+  private _policy: Policy;
+  private _masterKeysCoverCrypt: AbeMasterKey;
+
+  constructor(db: DB, policyAxes: PolicyAxis[], maxAttributeCreations: number) {
+    this._db = db;
+    this._policy = new Policy(policyAxes, maxAttributeCreations)
+    this._masterKeysCoverCrypt = generateMasterKeys(this._policy);
+  }
+
+  public get db(): DB {
+    return this._db;
+  }
+  public get policy(): Policy {
+    return this._policy;
+  }
+  public get masterKeysCoverCrypt(): AbeMasterKey {
+    return this._masterKeysCoverCrypt;
+  }
+
+  async encryptUsers(metadataUid: string) {
+    const policyBytes = this._policy.toJsonEncoded();
+
+    const elements = await this._db.getUsers();
+    for (const element of elements) {
```

Comment:
Rename `elements` to `users` and `element` to `user`?

Reply:
indeed!